Cut 0.0.106 (#1525)
lightningdevkit/rust-lightning: CHANGELOG.md

   payments. This is configured for outbound channels using
   `ChannelHandshakeLimits::trust_own_funding_0conf` whereas
   `ChannelManager::accept_inbound_channel_from_trusted_peer_0conf` is used for
   accepting inbound channels (#1401, #1505).
 * `ChannelManager::claim_funds` no longer returns a `bool` to indicate success.
   Instead, an `Event::PaymentClaimed` is generated if the claim was successful.
   Likewise, `ChannelManager::fail_htlc_backwards` no longer has a return value
   (#1434).
 * `lightning-rapid-gossip-sync` is a new crate for syncing gossip data from a
   server, primarily aimed at mobile devices (#1155).
 * `RapidGossipSync` can be passed to `BackgroundProcessor` in order to persist
   the `NetworkGraph` and handle `NetworkUpdate`s during event handling (#1433,
   #1517).
 * `NetGraphMsgHandler` has been renamed to `P2PGossipSync`, the `network_graph`
    module has been renamed to `gossip`, and `NetworkUpdate::ChannelClosed` has
   been renamed `NetworkUpdate::ChannelFailure` (#1159).
 * Added a `filtered_block_connected` method to `chain::Listen` and a default
   implementation of `block_connected` in order to support BIP 157/158 compact

>    block filters (#1453).

Author: Changed though note for compact block filters you still need to connect empty blocks if the filter doesn't match.

 * `InvoicePayer::new` now takes a `Retry` enum (replacing the `RetryAttempts`
   struct), which supports both attempt- and timeout-based retrying (#1418).
 * `Score::channel_penalty_msat` takes a `ChannelUsage` struct, which contains
   the capacity as an `EffectiveCapacity` enum and any potential in-flight HTLC
   value, rather than a single `u64`. Used by `ProbabilisticScorer` for more
   accurate penalties (#1456).
 * `build_route_from_hops` is a new function useful for constructing a route to
   probe (#1491).
 * `FundingLocked` message has been renamed `ChannelReady`, and related
   identifiers have been renamed accordingly (#1506).
 * `core2::io` or `std::io` (depending on feature flags `no-std` or `std`) is
   exported as a `lightning::io` module (#1504).
 * The deprecated `Scorer` has been removed in favor or `ProbabilisitcScorer`
   (#1512).

## Performance Improvements
 * `lightning-persister` crates uses `BufWriter` for improved performance
   (#1404).
 * Log gossip query messages at `GOSSIP` instead of `TRACE` to avoid
   overwhelming default logging (#1421).
 * `PeerManager` supports processing messages from different peers in parallel
   (#1023).

>  * `PeerManager` uses a single `Secp256k1` across all peers (#1472).

Author: Updated but let me know if you prefer to drop it.

## Bug Fixes
 * `ProabilisticScorer` uses more precision when approximating `log10` (#1406).
 * `BackgroundProcessor` process pending events before processing network
   messages to reduce lock contention in `ChannelManager`, which can be
   problematic for extremely slow machines (#1436).

>  * Fixed an integer underflow during channel opening occurring when the channel

Author: Dropped

# 0.0.107 - 2022-06-07

## API Updates

>  * Wumbo channels are now supported and can be enabled for inbound channels

Author: Hmmm... not sure. Figured given everyone else has supported it for ages, it's quite notable that we now do, too. Happy to push this down if you prefer.

   using `ChannelHandshakeLimits::max_funding_satoshis` (#1425).
 * Support for feature `option_zeroconf`, allowing immediate forwarding of
   payments. This is configured for outbound channels using
   `ChannelHandshakeLimits::trust_own_funding_0conf` whereas
   `ChannelManager::accept_inbound_channel_from_trusted_peer_0conf` is used for
   accepting inbound channels (#1401, #1505).
 * `ChannelManager::claim_funds` no longer returns a `bool` to indicate success.
   Instead, an `Event::PaymentClaimed` is generated if the claim was successful.
   Likewise, `ChannelManager::fail_htlc_backwards` no longer has a return value

>    (#1434).

Author: I'm more inclined to keep this brief and allow users to complain if they find it confusing.

 * `lightning-rapid-gossip-sync` is a new crate for syncing gossip data from a
   server, primarily aimed at mobile devices (#1155).
 * `RapidGossipSync` can be passed to `BackgroundProcessor` in order to persist
   the `NetworkGraph` and handle `NetworkUpdate`s during event handling (#1433,
   #1517).
 * `NetGraphMsgHandler` has been renamed to `P2PGossipSync`, the `network_graph`
    module has been renamed to `gossip`, and `NetworkUpdate::ChannelClosed` has
   been renamed `NetworkUpdate::ChannelFailure` (#1159).
 * Added a `filtered_block_connected` method to `chain::Listen` and a default
   implementation of `block_connected` for those fetching filtered instead of
   full blocks (#1453).
 * The `lightning-block-sync` crate's `BlockSource` trait methods now take
   `&self` instead of `&mut self` (#1307).
 * `inbound_payment` module is now public to allow for creating phantom invoices
   without a `ChannelManager` (#1384).
 * `lightning-block-sync`'s `init` and `poll` modules support `&dyn BlockSource`
   which can be determined at runtime (#1423).
 * `lightning-invoice` crate's `utils` now accept an expiration time (#1422,
   #1474).

>  * `find_route` now assumes variable-length onions by default (#1414).

Author: Made a spec compliance section.

 * `Event::PaymentForwarded` includes `prev_channel_id` and `next_channel_id`
   (#1419, #1475).
 * `chain::Watch::release_pending_monitor_events`' return type now associates
   `MonitorEvent`s with funding `OutPoints` (#1475).
 * `lightning-background-processor` crate's `Persister` trait has been moved to
   `lightning` crate's `util::persist` module, which now has a general
   `KVStorePersister` trait. Blanket implementations of `Persister` and
   `chainmonitor::Persist` are given for types implementing `KVStorePersister`.
   ` lightning-persister`'s `FilesystemPersister` implements `KVStorePersister`
   (#1417).

>  * `ChannelDetails` and `ChannelCounterparty` include fields for HTLC minimum

Reviewer: > Yea, 1444 is unrelated here - it's a bugfix to set the value correctly. The new config knob is in the next bullet.

Nice, thanks for adding the config knob as a separate bullet @jkczyz. This is now resolved after 1444 is removed from this bullet :)

   and maximum values (#1378, #1444).
 * `ProabilisticScorer` is parameterized by a `Logger`, which it uses to log
   channel liquidity updates or lack thereof (#1405).

>  * `ChannelDetails` has a `outbound_htlc_limit_msat` field, which should be used

Reviewer 2: Nit

```suggestion
 * `ChannelDetails` has an `outbound_htlc_limit_msat` field, which should be used
```

   in routing instead of `outbound_capacity_msat` (#1435).
 * `ProabilisticScorer`'s channel liquidities can be logged via
   `debug_log_liquidity_stats` (#1460).
 * `BackgroundProcessor` now takes an optional `WriteableScore` which it will
   persist using the `Persister` trait's new `persist_scorer` method (#1416).

>  * A `warn` message is now sent when receiving a `channel_reestablish` with an

Author: Moved to new spec section.

   old commitment transaction number rather than immediately force-closing the
   channel (#1430).
 * Upgraded to `bitcoin` crate version 0.28.1 (#1389).
 * `ShutdownScript::new_witness_program` now takes a `WitnessVersion` instead of
   a `NonZeroU8` (#1389).

>  * When a `channel_update` message is included in an onion error's `failuremsg`,

Author: Placed it in the spec compliance section and expanded to include reading. Let me know if you prefer the latter go under bug fixes.

   when a peer is disconnected prior to completing funding. Thus, channels will
   no longer be automatically force closed when the counterparty is disconnected
   due to incompatibility (#1429).
 * `Event::FundingGenerationReady` now has a `counterparty_node_id` field, which
   is used to pass to `ChannelManager` methods for funding, accepting, and
   closing the channel (#1479, #1485).
 * `InvoicePayer::new` now takes a `Retry` enum (replacing the `RetryAttempts`
   struct), which supports both attempt- and timeout-based retrying (#1418).
 * `Score::channel_penalty_msat` takes a `ChannelUsage` struct, which contains
   the capacity as an `EffectiveCapacity` enum and any potential in-flight HTLC
   value, rather than a single `u64`. Used by `ProbabilisticScorer` for more
   accurate penalties (#1456).
 * `build_route_from_hops` is a new function useful for constructing a route to
   probe (#1491).
 * `FundingLocked` message has been renamed `ChannelReady`, and related
   identifiers have been renamed accordingly (#1506).
 * `core2::io` or `std::io` (depending on feature flags `no-std` or `std`) is
   exported as a `lightning::io` module (#1504).

>  * The deprecated `Scorer` has been removed in favor or `ProbabilisitcScorer`

Reviewer 3: Nit: this typo still exists in the latest diff.

   (#1512).

## Performance Improvements
 * `lightning-persister` crate's `FilesystemPersister` is faster by 15x (#1404).
 * Log gossip query messages at `GOSSIP` instead of `TRACE` to avoid
   overwhelming default logging (#1421).
 * `PeerManager` supports processing messages from different peers in parallel,
   and this is taken advantage of in gossip processing (#1023).
 * `PeerManager` uses a single `Secp256k1` across all peers for reduced memory
   usage (#1472).

## Bug Fixes
 * Fixed a bug where crashing while persisting a `ChannelMonitorUpdate` for a
   part of a multi-path payment would cause loss of funds due to a partial
   payment claim on restart (#1434).

>  * `ProabilisticScorer` uses more precision when approximating `log10` (#1406).

Author: Mainly had just listed in merge order then moved notable API changes to the top. Moved this to the end now.

 * `BackgroundProcessor` processes pending events before processing network
   messages to reduce lock contention in `ChannelManager`, which improves
   `ChannelManager` serialization on slow systems to avoid force channel closes
   (#1436).
 * `gossip_timestamp_filter` filters are now honored when sending gossip to
   peers (#1452).
 * During a reorg, only force-close a channel if its funding transaction is
   unconfirmed rather than as it loses confirmations (#1461).
 * Fixed a panic in `lightning-net-tokio` when fetching a peer's socket address
   after the connection has been closed (#1449).

>  * `find_route` will no longer return routes that would cause Onion construction

Reviewer 2: nit: common noun?

```suggestion
 * `find_route` will no longer return routes that would cause onion construction
```

   channel (#1430).
 * When a `channel_update` message is included in an onion error's `failuremsg`,
   its message type is now encoded. Reading such messages is also supported
   (#1465).

## Bug Fixes
 * Fixed a bug where crashing while persisting a `ChannelMonitorUpdate` for a
   part of a multi-path payment could cause loss of funds due to a partial
   payment claim on restart (#1434).
 * `BackgroundProcessor` has been fixed to improve serialization reliability on
   slow systems which can avoid force-closes (#1436).
 * `gossip_timestamp_filter` filters are now honored when sending gossip to
   peers (#1452).
 * During a reorg, only force-close a channel if its funding transaction is
   unconfirmed rather than as it loses confirmations (#1461).
 * Fixed a rare panic in `lightning-net-tokio` when fetching a peer's socket
   address after the connection has been closed caused by a race condition
   (#1449).
 * `find_route` will no longer return routes that would cause onion construction
   to fail in some cases (#1476).
 * `ProbabilisticScorer` uses more precision when approximating `log10` (#1406).

## Serialization Compatibility

> * `ChannelManager` serialization is no longer compatible with versions prior to

Author: Is this for both `ChannelManager::accept_inbound_channel_from_trusted_peer_0conf` and `ChannelHandshakeLimits::trust_own_funding_0conf`? Should this be called out in their respective docs?

> * `ChannelManager` serialization is no longer compatible with versions prior to

Author: Might have misread this. Was assuming you meant whenever `Channel::set_0conf`. Was this comment specific about features set on the channel? Let me know if there's a phrasing that you would prefer.

> * `ChannelManager` serialization is no longer compatible with versions prior to

Author: > It's kinda independent - we never set the feature, but still do 0conf. Inbound channel peers may set the feature, which would cause incompatibility.

Don't we set as optional in `InitFeatures::known`?

BTW, pushed wording essentially what you said in the first comment. Let me know if you prefer different wording?

# 0.0.107 - 2022-06-08

## API Updates

>  * Wumbo channels are now supported and can be enabled for inbound channels

Reviewer 4: ```suggestion
 * Channels larger than 16777215 sats (Wumbo!) are now supported and can be enabled for inbound channels
```

   using `ChannelHandshakeLimits::max_funding_satoshis` (#1425).
 * Support for feature `option_zeroconf`, allowing immediate forwarding of

>    payments. This is configured for outbound channels using

Reviewer 4: ```suggestion
   payments after channel opening. This is configured for outbound channels using
```

   `ChannelHandshakeLimits::trust_own_funding_0conf` whereas

>    `ChannelManager::accept_inbound_channel_from_trusted_peer_0conf` is used for

Reviewer 4: ```suggestion
   `ChannelManager::accept_inbound_channel_from_trusted_peer_0conf` has to be used for
```

   persist using the `Persister` trait's new `persist_scorer` method (#1416).
 * Upgraded to `bitcoin` crate version 0.28.1 (#1389).
 * `ShutdownScript::new_witness_program` now takes a `WitnessVersion` instead of
   a `NonZeroU8` (#1389).
 * Channels will no longer be automatically force closed when the counterparty
   is disconnected due to incompatibility (#1429).
 * `ChannelManager` methods for funding, accepting, and closing channels now
   take a `counterparty_node_id` parameter, which has also been added as a field
   to `Event::FundingGenerationReady` (#1479, #1485).
 * `InvoicePayer::new` now takes a `Retry` enum (replacing the `RetryAttempts`
   struct), which supports both attempt- and timeout-based retrying (#1418).
 * `Score::channel_penalty_msat` takes a `ChannelUsage` struct, which contains
   the capacity as an `EffectiveCapacity` enum and any potential in-flight HTLC
   value, rather than a single `u64`. Used by `ProbabilisticScorer` for more
   accurate penalties (#1456).
 * `build_route_from_hops` is a new function useful for constructing a `Route`
   given a specific list of public keys (#1491).

>  * `FundingLocked` message has been renamed `ChannelReady`, and related

Reviewer 4: Could be moved to `Spec Compliance`.

   identifiers have been renamed accordingly (#1506).
 * `core2::io` or `std::io` (depending on feature flags `no-std` or `std`) is
   exported as a `lightning::io` module (#1504).
 * The deprecated `Scorer` has been removed in favor or `ProbabilisitcScorer`
   (#1512).

## Performance Improvements
 * `lightning-persister` crate's `FilesystemPersister` is faster by 15x (#1404).
 * Log gossip query messages at `GOSSIP` instead of `TRACE` to avoid
   overwhelming default logging (#1421).
 * `PeerManager` supports processing messages from different peers in parallel,
   and this is taken advantage of in gossip processing (#1023).
 * Reduced per-peer memory usage in `PeerManager` (#1472).

## Spec Compliance
 * `find_route` now assumes variable-length onions by default for nodes

>    supporting the feature (#1414).

Author: Ah, misinterpreted your earlier comment.

   path with an overall lower cost (#1399).

## Serialization Compatibility

>  * All above new events/fields are ignored by prior clients. All above new

Author: Oops, removed.
